### PR TITLE
[Mission] CoP6-2 Mechanics

### PR DESCRIPTION
**Pull request by The-Aerec**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

Adds shared ability usage between the three NMs depending on situation.

## Steps to test these changes

!addmission 6 618
!pos 260 9 -435 25

Fight mobs -> Watch sync'd abilities and ability responses based on enfeebles.

### CODE REVIEW

File: scripts/globals/mobskills/reactive_armor.lua [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---------------------------------------------
+-- Reactive Armor
+-- Description: Covers the user in shock spikes. Enemies that hit it take lightning damage.
+-- Type: Enhancing
+-- Utsusemi/Blink absorb: N/A
+-- Range: Self
+-- Notes:
+---------------------------------------------
+require("scripts/globals/mobskills")
+require("scripts/settings/main")
```
Comment (claywar) [inline scripts/globals/mobskills/reactive_armor.lua]: Settings does not appear to be used here, and should not be required if it isn't needed

File: scripts/mixins/warders_cop.lua [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+-- Mixin for Warder NMs in CoP
+-----------------------------------
+require('scripts/globals/mixins')
+require('scripts/globals/status')
+-----------------------------------
+
+g_mixins = g_mixins or {}
+
+g_mixins.warders_cop = function(mob)
+
```
Comment (claywar) [inline scripts/mixins/warders_cop.lua]: Can remove the newline here

File: scripts/mixins/warders_cop.lua [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    mob:addListener('COMBAT_TICK', 'WARDER_CTICK', function(mob)
+        local battletime = mob:getBattleTime()
+        local changeTime = mob:getLocalVar('changeTime')
+        local initiate = mob:getLocalVar('initiate')
+        local warder = mob:getLocalVar('warder')
+        local electro = mob:getLocalVar('electro')
+        local reactive = mob:getLocalVar('reactive')
+        local ID = zones[mob:getZoneID()]
+
+        if battletime - initiate >= 27 and
```
Comment (claywar) [inline scripts/mixins/warders_cop.lua]: Styling, if conditions are broken out, it should follow:
```
if
    condition1 and
    condition2
then
```

File: scripts/mixins/warders_cop.lua [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+           warder <= 1
+        then -- warder signals to the others before using abilities
+            mob:showText(mob, ID.text.BEEP_BEEP)
+            mob:setLocalVar('initiate', battletime)
+        elseif battletime - changeTime >= 30 and
```
Comment (claywar) [inline scripts/mixins/warders_cop.lua]: Same as above

File: scripts/zones/Misareaux_Coast/IDs.lua [hunk 5]
```diff
@@ -24,6 +24,9 @@ zones[xi.zone.MISAREAUX_COAST] =
         CONQUEST_BASE            = 7179, -- Tallying conquest results...
         DOOR_CLOSED              = 7355, -- The door is locked tight.
         SNOWMINT_POINT_LOCKED    = 7358, -- This gate leads to Snowmint Point. However, it seems to be locked...
+        BEEP_BEEP                = 7580, -- Beep...beep...beep...beep...
```
Comment (claywar) [inline scripts/zones/Misareaux_Coast/IDs.lua:27]: 👍 Looks correct for IDs with current client

File: scripts/globals/mobskills/reactive_armor.lua [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+---------------------------------------------
+local mobskill_object = {}
+
+mobskill_object.onMobSkillCheck = function(target, mob, skill)
+    return 0
+end
+
+mobskill_object.onMobWeaponSkill = function(target, mob, skill)
+    local power = math.random(20, 30)
+    local duration = 180
```
Comment (claywar) [inline scripts/globals/mobskills/reactive_armor.lua:20]: Unused var

File: scripts/mixins/warders_cop.lua [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+                mob:useMobAbility(561) -- use Electromagnetic Field
+                mob:setLocalVar('electro', 0)
+            elseif reactive == 1 then
+                mob:useMobAbility(562) -- use Reactive Armor
+                mob:setLocalVar('reactive', 0)
+            end
+        end
+    end)
+
+    mob:addListener('WEAPONSKILL_STATE_ENTER', 'WS_START_MSG', function(mob, skillID)
```
Comment (claywar) [inline scripts/mixins/warders_cop.lua]: `function(mob, skillID)` - This shadows the mob parameter of the original function.  To resolve, the pattern we've been using for the lower function is to change it to `function(mobArg, skillID)` and replace the listener function references of mob to mobArg

**Review by claywar — APPROVED** [review]

**Comment by claywar** [comment]
Conflict here that needs to be resolved in the sql

**Comment by The-Aerec** [comment]
> Conflict here that needs to be resolved in the sql

Fixed

**Pull request merged**